File: app/src/main/java/com/wojdor/hearthcards/application/card/CardViewModel.java
package com.wojdor.hearthcards.application.card;

import android.app.Application;
import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.support.annotation.NonNull;

import com.wojdor.hearthcards.application.base.BaseAndroidViewModel;
import com.wojdor.hearthcards.application.util.FileStorage;
import com.wojdor.hearthcards.domain.Card;

import java.io.File;

public class CardViewModel extends BaseAndroidViewModel {

    private FileStorage fileStorage;

    public CardViewModel(@NonNull Application application) {
        super(application);
        this.fileStorage = new FileStorage(application);
    }

    public LiveData<Card> getCardByCardId(String cardId) {
        return cardDao.getCardByCardId(cardId);
    }

    public LiveData<File> getCardImage(String cardId) {
        MutableLiveData<File> data = new MutableLiveData<>();
        File file = fileStorage.get(cardId);
        data.postValue(file);
        return data;
    }
}

File: app/src/main/java/com/wojdor/hearthstonecards/app/card/CardViewModel.kt
package com.wojdor.hearthstonecards.app.card

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.wojdor.hearthstonecards.app.base.BaseViewModel
import com.wojdor.hearthstonecards.data.repository.CardRepository
import com.wojdor.hearthstonecards.domain.Card
import java.io.File

class CardViewModel(repository: CardRepository) : BaseViewModel(repository) {

    fun getCardByCardId(cardId: String): LiveData<Card> = repository.getCardByCardId(cardId)

    fun getCardImage(cardId: String): LiveData<File> {
        val file = repository.getCardImageFromStorage(cardId)
        return MutableLiveData<File>().apply { postValue(file) }
    }
}
